docs/behat/basic-usage.rst

```diff
@@ -0,0 +1,540 @@
+
+Usage example of ``->getElement('your_element')`` and ``->getDefinedElements`` methods.
+
+.. code-block:: php
+
+    <?php
+
+    class CreatePage extends SymfonyPage implements CreatePageInterface
+    {
+        // This method returns a simple associative array, where the key is the name of your element and the value is its locator.
+        protected function getDefinedElements()
+        {
+            return array_merge(parent::getDefinedElements(), [
+                'provinces' => '#sylius_country_provinces',
+            ]);
+        }
+
+        // By default it will assume that your locator is css.
+        // Example with xpath.
+        protected function getDefinedElements()
+        {
+            return array_merge(parent::getDefinedElements(), [
+                'provinces' => ['xpath' => '//*[contains(@class, "provinces")]'] // Now your value is an array where key is your locator type.
+            ]);
+        }
+
+        // Like that you can easily manipulate your page elements.
+        public function addProvince(ProvinceInterface $province)
+        {
+            $provinceSelectBox = $this->getElement('provinces');
+
+            $provinceSelectBox->selectOption($province->getName());
+        }
+    }
+
+Let's get back to our main example and analyze our scenario.
+We have steps like
+"When I choose "France"
+And I add it
+Then I should be notified that it has been successfully created
+And the country "France" should appear in the store".
+
+.. code-block:: php
+
+    <?php
+
+    namespace Sylius\Behat\Page\Admin\Country;
+
+    use Sylius\Behat\Page\Admin\Crud\CreatePage as BaseCreatePage;
+
+    class CreatePage extends BaseCreatePage implements CreatePageInterface
+    {
+        /**
+         * @param string $name
+         */
+        public function chooseName($name)
+        {
+            $this->getDocument()->selectFieldOption('Name', $name);
+        }
+
+        public function create()
+        {
+            $this->getDocument()->pressButton('Create');
+        }
+    }
+
+.. code-block:: php
+
+    <? php
+
+    namespace Sylius\Behat\Page\Admin\Country;
+
+    use Sylius\Behat\Page\Admin\Crud\IndexPage as BaseIndexPage;
+
+    class IndexPage extends BaseIndexPage implements IndexPageInterface
+    {
+        /**
+         * @return bool
+         */
+        public function isSingleResourceOnPage(array $parameters)
+        {
+            try {
+                $rows = $this->tableAccessor->getRowsWithFields($this->getElement('table'), $parameters);
+                // Table accessor is a helper service which is responsible for all html table operations.
+
+                return 1 === count($rows);
+            } catch (\InvalidArgumentException $exception) {
+                return false;
+            } catch (ElementNotFoundException $exception) {
+                return false;
```

Comment:
I would add a comment that we catch `ElementNotFoundException` since if there are no resources the `table` is not rendered (both in the docs and the code 😄 ).